x/auth/vesting/internal/types/vesting_account.go

```diff
@@ -0,0 +1,726 @@
+// vesting, nil is returned.
+func (cva ContinuousVestingAccount) GetVestingCoins(blockTime time.Time) sdk.Coins {
+	return cva.OriginalVesting.Sub(cva.GetVestedCoins(blockTime))
+}
+
+// SpendableCoins returns the total number of spendable coins per denom for a
+// continuous vesting account.
+func (cva ContinuousVestingAccount) SpendableCoins(blockTime time.Time) sdk.Coins {
+	return cva.spendableCoins(cva.GetVestingCoins(blockTime))
+}
+
+// TrackDelegation tracks a desired delegation amount by setting the appropriate
+// values for the amount of delegated vesting, delegated free, and reducing the
+// overall amount of base coins.
+func (cva *ContinuousVestingAccount) TrackDelegation(blockTime time.Time, amount sdk.Coins) {
+	cva.trackDelegation(cva.GetVestingCoins(blockTime), amount)
+}
+
+// GetStartTime returns the time when vesting starts for a continuous vesting
+// account.
+func (cva *ContinuousVestingAccount) GetStartTime() int64 {
+	return cva.StartTime
+}
+
+// GetEndTime returns the time when vesting ends for a continuous vesting account.
+func (cva *ContinuousVestingAccount) GetEndTime() int64 {
+	return cva.EndTime
+}
+
+// Validate checks for errors on the account fields
+func (cva ContinuousVestingAccount) Validate() error {
+	if cva.GetStartTime() >= cva.GetEndTime() {
+		return errors.New("vesting start-time cannot be before end-time")
+	}
+
+	return cva.BaseVestingAccount.Validate()
+}
+
+// MarshalYAML returns the YAML representation of an account.
+func (cva ContinuousVestingAccount) MarshalYAML() (interface{}, error) {
+	var bs []byte
+	var err error
+	var pubkey string
+
+	if cva.PubKey != nil {
+		pubkey, err = sdk.Bech32ifyAccPub(cva.PubKey)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	bs, err = yaml.Marshal(struct {
+		Address          sdk.AccAddress
+		Coins            sdk.Coins
+		PubKey           string
+		AccountNumber    uint64
+		Sequence         uint64
+		OriginalVesting  sdk.Coins
+		DelegatedFree    sdk.Coins
+		DelegatedVesting sdk.Coins
+		EndTime          int64
+		StartTime        int64
+	}{
+		Address:          cva.Address,
+		Coins:            cva.Coins,
+		PubKey:           pubkey,
+		AccountNumber:    cva.AccountNumber,
+		Sequence:         cva.Sequence,
+		OriginalVesting:  cva.OriginalVesting,
+		DelegatedFree:    cva.DelegatedFree,
+		DelegatedVesting: cva.DelegatedVesting,
+		EndTime:          cva.EndTime,
+		StartTime:        cva.StartTime,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return string(bs), err
+}
+
+//-----------------------------------------------------------------------------
+// Periodic Vesting Account
+
+var _ vestexported.VestingAccount = (*PeriodicVestingAccount)(nil)
+var _ authexported.GenesisAccount = (*PeriodicVestingAccount)(nil)
+
+// PeriodicVestingAccount implements the VestingAccount interface. It
+// periodically vests by unlocking coins during each specified period
+type PeriodicVestingAccount struct {
```

Comment:
Is this supposed to be the "custom vesting schedule" type? What is `PeriodicVestingAccount`? I don't see how this is custom. At the very least it should be added to the spec to describe how and why it differs from the others.

Ulimately, we need a generic/custom vesting account type that allows you to define your own blockbox logic for `GetVestedCoins` and `GetVestingCoins`.

Reply:
Update the specification document to include `PeriodicVestingAccount`

> I don't see how this is custom.

I interpret custom to mean 'coins vest on a schedule that I define, both with respect to time and amount'. A periodic vesting account allows that. The structure is general enough that any function (tokens over time) can be approximated. So, unlike `ContinuousVestingAccount`, it can do quarterly vesting, yearly vesting, or weird things like exponential vesting.  I think the main advantage for end users is that `PeriodicVestingAccount` can release coins in staggered tranches rather than continuously, which is a common deal structure for investors/founders/advisors. If there is some other type of customization that is being sought, please clarify further. 

> Ulimately, we need a generic/custom vesting account type that allows you to define your own blockbox logic for GetVestedCoins and GetVestingCoins.

The VestingAccount interface allows for custom logic in `GetVestedCoins` and `GetVestingCoins`. I don't think it needs to be changed to enable what you are suggesting.

Reply:
Can you update the spec with these details and even examples?